### PR TITLE
Fix publish data not being cleared when removing last non-deleted iteration

### LINKED ISSUE #6231: Some exercises don't count as completed while they are completed
In my C# track, I have 0 available exercises while 105 are still locked. I completed all my available exercises:

It seems that some completed exercises don't count as such. If I take a look at the _locked_ **Accumulate** exercise, it needs the completion of 2 exercises to be unlocked: **Tracks on Tracks on Tracks** and **Log Analysis**. 
If I compare the main page of these two exercises, I got the following:

While **Log Analysis** displays that it is completed, it is not the case for **Tracks on Tracks on Tracks**. Maybe that's part of the problem?

@ErikSchierboom maybe you have an idea of what's causing the trouble here?

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/exercism/issues/6231

### CODE REVIEW

File: app/commands/iteration/destroy.rb [hunk 1]
```diff
@@ -7,14 +7,15 @@ class Destroy
     def call
       iteration.update!(deleted_at: Time.current)
 
-      # Unpublish the solution if this was the published iteration
-      solution.update!(published_iteration_id: nil, published_at: nil) if solution.published_iteration_id == iteration.id
-
-      # Uncomplete the solution if there are now no iterations
-      solution.update!(completed_at: nil) unless solution.iterations.not_deleted.exists?
+      if !solution.iterations.not_deleted.exists?
```
Comment (iHiD) [inline app/commands/iteration/destroy.rb:10]:
```suggestion
      unless solution.iterations.not_deleted.exists?
```

File: app/commands/solution/uncomplete.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Solution
+  class Uncomplete
+    include Mandate
+
+    initialize_with :solution
+
+    def call
+      solution.update!(completed_at: nil)
+      Solution::Unpublish.(solution)
```
Comment (iHiD) [inline app/commands/solution/uncomplete.rb]: Maybe wrap this in a Transaction?

solution.transaction do
...
end

File: app/commands/iteration/destroy.rb [hunk 3]
```diff
@@ -7,14 +7,15 @@ class Destroy
...
+        Solution::Uncomplete.(solution)
+      elsif solution.published_iteration_id == iteration.id
+        Solution::Unpublish.(solution)
+      end
```
Comment (iHiD) [inline app/commands/iteration/destroy.rb]: How about this to make it more readable. The second bit might run twice, but will just be a no-op the second time so it's not really painful.

```suggestion
      Solution::Uncomplete.(solution) unless solution.iterations.not_deleted.exists?
      Solution::Unpublish.(solution) if solution.published_iteration_id == iteration.id
```

**Comment by ErikSchierboom** [comment]
> The second bit might run twice, but will just be a no-op the second time so it's not really painful.

I had that originally but reverted for the above-mentioned duplicate run. But if it is no big deal, that is much cleaner.

**Pull request merged**